mobile/lib/services/sync.service.dart

```diff
@@ -402,32 +406,31 @@ class SyncService {
     album.lastModifiedAssetTimestamp = originalDto.lastModifiedAssetTimestamp;
     album.shared = dto.shared;
     album.activityEnabled = dto.activityEnabled;
-    if (album.thumbnail.value?.remoteId != dto.remoteThumbnailAssetId) {
-      album.thumbnail.value = await _db.assets
-          .where()
-          .remoteIdEqualTo(dto.remoteThumbnailAssetId)
-          .findFirst();
+    final remoteThumbnailAssetId = dto.remoteThumbnailAssetId;
+    if (remoteThumbnailAssetId != null &&
+        album.thumbnail.value?.remoteId != remoteThumbnailAssetId) {
+      album.thumbnail.value =
+          await _assetRepository.getByRemoteId(remoteThumbnailAssetId);
     }
 
     // write & commit all changes to DB
     try {
-      await _db.writeTxn(() async {
-        await _db.assets.putAll(toUpdate);
-        await album.thumbnail.save();
-        await album.sharedUsers
-            .update(link: usersToLink, unlink: usersToUnlink);
-        await album.assets.update(link: assetsToLink, unlink: toUnlink.cast());
-        await _db.albums.put(album);
-      });
+      await _assetRepository.updateAll(toUpdate);
+      await _albumRepository.addUsers(album, usersToLink);
+      await _albumRepository.removeUsers(album, usersToUnlink);
+      await _albumRepository.addAssets(album, assetsToLink);
+      await _albumRepository.removeAssets(album, toUnlink);
+      await _albumRepository.recalculateMetadata(album);
+      await _albumRepository.update(album);
```

Comment:
Does it matter that these are individual transactions?

Reply:
it should not as the album is updated last (only if everything beforehand is ok). in any case, I've reworked all occasions to use transactions where applicable as it results in less re-renderings of the UI